File: api/src/main/java/eu/suro/grpc/GRPChannel.java
package eu.suro.grpc;

import de.exlll.configlib.YamlConfigurations;
import eu.suro.utils.Log;
import io.grpc.ChannelCredentials;
import io.grpc.Grpc;
import io.grpc.InsecureChannelCredentials;
import io.grpc.ManagedChannel;


import java.io.File;
import java.nio.file.Path;

public class GRPChannel {

    private static ManagedChannel channel;

    public static void init(File dataFolder) {
        if(channel == null){
            File configFile = new File(dataFolder, "config.yml");
            GRPCConfig grpcConfig = new GRPCConfig();
            if(!configFile.exists()){
                YamlConfigurations.save(configFile.toPath(), GRPCConfig.class, grpcConfig);
            }
            grpcConfig = YamlConfigurations.load(configFile.toPath(), GRPCConfig.class);
            ChannelCredentials credentials = InsecureChannelCredentials.create();
            GRPChannel.channel = Grpc.newChannelBuilder(grpcConfig.getHost()+":"+grpcConfig.getPort(),credentials)
                    .build();
            Log.info("[GRPC] Запускаем канал связи с ядром");
        }
    }

    public static ManagedChannel getChannel() {
        return channel;
    }
}

File: api/src/main/kotlin/eu/suro/grpc/GRPChannel.kt
package eu.suro.grpc

import de.exlll.configlib.YamlConfigurations
import io.grpc.Grpc
import io.grpc.InsecureChannelCredentials
import io.grpc.ManagedChannel
import java.io.File

class GRPChannel {
    /**
     * Создание канала для подключения к серверу
     * @param dataFolder - Папка с конфигом
     */
    companion object{
        var channel: ManagedChannel? = null
        private set

        @JvmStatic fun init(dataFolder: File){
            if(channel == null){
                val configFile = File(dataFolder, "config.yml")
                if(!configFile.exists()){
                    YamlConfigurations.save(configFile.toPath(), GRPCConfig::class.java, GRPCConfig())
                }
                val grpcConfig: GRPCConfig = YamlConfigurations.load(configFile.toPath(), GRPCConfig::class.java)
                val credentials = InsecureChannelCredentials.create()
                println(grpcConfig.host+":"+grpcConfig.port)
                channel = Grpc.newChannelBuilder(grpcConfig.host + ":" + grpcConfig.port, credentials).build()
                println(channel)
                println("[GRPC] Запускаем канал связи с ядром")
            }
        }
    }
}